src/empire/object.c

```diff
@@ -526,7 +526,8 @@ int empire_object_city_buys_resource(int object_id, int resource)
 int empire_object_city_sells_resource(int object_id, int resource)
 {
     const full_empire_object *object = array_item(objects, object_id);
-    if (object->city_sells_resource[resource]) {
+    if (object->city_sells_resource[resource] &&
+        (resource_is_storable(resource) || object->city_type == EMPIRE_CITY_OURS)) {
```

Comment:
Why ` || object->city_type == EMPIRE_CITY_OURS` ?
Wouldn't this make it possible to 'sell' concrete even though the traders cannot ever pick it up from a warehouse?